bitshuffle/bitshuffle.py

```diff
@@ -0,0 +1,445 @@
+    BitShuffle data packets.
+
+    :param fhandle:
+    """
+
+    try:
+        # Python 3
+        data = fhandle.buffer.read()
+    except Exception as e:
+        data = fhandle.read()
+    checksum = hashlib.sha1(data).hexdigest()
+    chunks = encode_data(data, chunksize, compresslevel, compresstype)
+    seqmax = len(chunks) - 1
+    seqnum = 0
+    packets = []
+    for c in chunks:
+        packets.append(encode_packet(c, checksum, seqnum,
+                                     seqmax, compresstype))
+        seqnum += 1
+
+    return packets
+
+
+def main():
+    parser = argparse.ArgumentParser(description="A tool for encoding and " +
+                                     "decoding arbitrary binary data to " +
+                                     "ASCII text.")
+
+    parser.add_argument("--input", "-i",
+                        help="Input file. Defaults to stdin. If the only " +
+                        "argument, implies --encode")
+
+    parser.add_argument("--output", "-o",
+                        help="Output file. Defaults to stdout. If the only " +
+                        "argument, and stdin is not a tty, implies " +
+                        "--decode")
+
+    parser.add_argument("--encode", "-e", action="store_true",
+                        help="Generate a BitShuffle data packet from" +
+                             "the input file and write it to the output.")
+
+    parser.add_argument("--decode", "-d", "-D", action="store_true",
+                        help="Extract BitShuffle data packet(s) from the " +
+                        "the input file, and write the decoded file to the " +
+                        "output file.")
+
+    parser.add_argument("--version", "-v", action="store_true",
+                        help="Displays the current version of bitshuffle.")
+
+    parser.add_argument("--chunksize", "-c", type=int,
+                        help="Chunk size in bytes. Defaults to 2048.")
+
+    parser.add_argument("--compresslevel", '-m', type=int,
+                        help="Compression level when encoding. " +
+                        "1 is lowest, 9 is highest. Defaults to 5. " +
+                        "Ignore if specified compresstype does not support" +
+                        " multiple compression levels.")
+
+    parser.add_argument("--editor", "-E",
+                        help="Editor to use for pasting packets. " +
+                        "If not specified, defaults in this order:\n" +
+                        "\t$VISUAL, $EDITOR, mimeopen, nano, vi, " +
+                        "emacs, micro, notepad, notepad++")
+
+    parser.add_argument("--compresstype", '-t',
+                        help="Type of compression to use. Defaults to bz2. " +
+                             "Ignored if decoding packets. " +
+                             "Currently supported: 'bz2', 'gzip'.")
+
+    args = parser.parse_args()
+
+    # Checks if no parameters were passed
+    if len(sys.argv[1:]) == 0:
```

Comment:
`if not sys.argv[1:]`

Reply:
fixed in 917686c

Reply:
Actually, I could have left it with `len()` in retrospect, as it would do the same thing.